store/mockstore/unistore/cophandler/closure_exec.go

```diff
@@ -62,6 +62,39 @@ func mapPkStatusToHandleStatus(pkStatus int) tablecodec.HandleStatus {
 	return tablecodec.HandleDefault
 }
 
+func getExecutorList(dagReq *tipb.DAGRequest) ([]*tipb.Executor, error) {
+	if len(dagReq.Executors) > 0 {
+		return dagReq.Executors, nil
+	}
+	// convert TiFlash executors tree to executor list
+	executors := make([]*tipb.Executor, 0, 3)
+	currentExec := dagReq.RootExecutor
+	for currentExec.Tp != tipb.ExecType_TypeTableScan {
+		executors = append(executors, currentExec)
+		switch currentExec.Tp {
+		case tipb.ExecType_TypeTopN:
+			currentExec = currentExec.TopN.Child
+		case tipb.ExecType_TypeStreamAgg, tipb.ExecType_TypeAggregation:
+			currentExec = currentExec.Aggregation.Child
+		case tipb.ExecType_TypeLimit:
+			currentExec = currentExec.Limit.Child
+		default:
+			return nil, errors.New("unknown supported executor type " + currentExec.Tp.String())
+		}
+	}
+	executors = append(executors, currentExec)
+	i := 0
+	j := len(executors) - 1
+	for i < j {
+		e := executors[i]
+		executors[i] = executors[j]
+		executors[j] = e
+		i++
+		j--
+	}
```

Comment:
for i, j := 0, len(executors)-1; i < j; i, j = i+1, j-1 {
    executors[i], executors[j] = executors[j], executors[i]
}